### PR TITLE
Node pipeline

### LINKED ISSUE #6402: 'node:' module prefix causes "Unknown pipeline: node" errors
# 🐛 bug report

'Came upon this issue while trying to use the `gravatar-url` module. (issue originally reported here: https://github.com/sindresorhus/gravatar-url/issues/8)

This works...
```
$ cat /tmp/index.html
<script type="module">
  import crypto from 'crypto';
</script>

$ parcel /tmp/index.html
ℹ️ Server running at http://localhost:49267
✨ Built in 486ms
```

This does not ...
```
$ cat /tmp/index.html
<script type="module">
  import crypto from 'node:crypto';
</script>

$ parcel /tmp/index.html
ℹ️ Server running at http://localhost:1234
🚨 Build failed.
@parcel/core: Unknown pipeline: node.
/private/tmp/index.html:1:20
> 1 | <script type="module">
>   |                    ^^^^
  2 |   import crypto from 'node:crypto';
  3 | </script>
```

The only difference here is the 'node:'-prefix on the module name in the second example.  Per https://nodejs.org/api/esm.html#esm_node_imports, this is a valid import name since node@14 and has been back-ported to node@12.

## 🌍 Your Environment

```
$ envinfo

  System:
    OS: macOS 11.4
    CPU: (8) x64 Apple M1
    Memory: 252.34 MB / 16.00 GB
    Shell: 5.1.0 - /usr/local/bin/bash
  Binaries:
    Node: 14.16.1 - ~/.nvm/versions/node/v14.16.1/bin/node
    npm: 7.15.0 - ~/.nvm/versions/node/v14.16.1/bin/npm
  Browsers:
    Chrome: 91.0.4472.77
    Firefox: 88.0.1
    Safari: 14.1.1
  npmPackages:
    @parcel/babel-plugin-transform-runtime: 2.0.0-nightly.2312 => 2.0.0-nightly.2312
    @parcel/packager-raw-url: 2.0.0-nightly.2312 => 2.0.0-nightly.2312
    @parcel/transformer-sass: 2.0.0-nightly.690 => 2.0.0-nightly.690
    @parcel/transformer-svg-react: 2.0.0-nightly.2312 => 2.0.0-nightly.2312
    @parcel/transformer-webmanifest: 2.0.0-nightly.2312 => 2.0.0-nightly.2312
    @types/react-dom: 17.0.3 => 17.0.3
    @types/react-router-dom: 5.1.7 => 5.1.7
    @typescript-eslint/eslint-plugin: 4.20.0 => 4.20.0
    @typescript-eslint/parser: 4.20.0 => 4.20.0
    blueimp-md5: 2.18.0 => 2.18.0
    bootstrap: 5.0.1 => 5.0.1
    checkcss: 1.0.2 => 1.0.2
    eslint: 7.21.0 => 7.21.0
    eslint-config-standard: 16.0.2 => 16.0.2
    eslint-plugin-import: 2.22.1 => 2.22.1
    eslint-plugin-node: 11.1.0 => 11.1.0
    eslint-plugin-promise: 4.3.1 => 4.3.1
    eslint-plugin-react: 7.22.0 => 7.22.0
    eslint-plugin-react-hooks: 4.2.0 => 4.2.0
    firebase: 8.4.2 => 8.4.2
    firebaseui: 4.8.0 => 4.8.0
    nanoid: 3.1.22 => 3.1.22
    parcel: 2.0.0-nightly.688 => 2.0.0-nightly.688
    react: ^16.13.1 => 16.14.0
    react-bootstrap: 2.0.0-alpha.2 => 2.0.0-alpha.2
    react-dom: ^16.13.1 => 16.14.0
    react-router-bootstrap: 0.25.0 => 0.25.0
    react-router-dom: 5.2.0 => 5.2.0
    sass: 1.32.12 => 1.32.12
    simplur: 3.0.1 => 3.0.1
    typescript: 4.2.3 => 4.2.3
```

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

This PR adds support for the `node:` prefix to explicitly define a node builtin

closes #6402 

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.50s | +1.00ms |
| Cached | 497.00ms | +21.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/logo.1e014c76.png | 274.00b | +0.00b | 310.00ms | +18.00ms ⚠️ |

</p></details>

<details><summary>AtlasKit Editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.43s | -74.00ms |
| Cached | 393.00ms | +3.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/302498019722396168)**

File: packages/core/core/src/requests/ParcelConfigRequest.js [hunk 1]
```diff
@@ -263,6 +263,17 @@ function processMap(
   // $FlowFixMe
   let res: ConfigMap<any, any> = {};
   for (let k in map) {
+    if (k.startsWith('node:')) {
+      throw new ThrowableDiagnostic({
+        diagnostic: {
+          message:
+            'Named pipeline node: is reserved for builtin NodeJS libraries',
```
Comment (mischnic) [inline packages/core/core/src/requests/ParcelConfigRequest.js]:
```suggestion
            'Named pipeline node: is reserved for builtin Node.js libraries',
```

**Comment by mischnic** [comment]
Should this be transpiled (= strip the `node:` part)? AFAICT, `require("node:crypto")` is only valid in Node 16 and only the ESM variant was backported to 12 and 14.

**Comment by DeMoorJasper** [comment]
@mischnic you're right it won't work without stripping of the `node:` part but if we just remove that it might behave differently compared to native node? (not entirely sure why they created this `node:` prefix and if they will deprecate importing without the prefix at some point)

Also in the case we transform it away it probably should be handled entirely in the transformer and blacklisted in the config loader without any changes to the resolver?

**Comment by mischnic** [comment]
> Also in the case we transform it away it probably should be handled entirely in the transformer and blacklisted in the config loader without any changes to the resolver?

Yeah, you're right. I'm not so sure myself anymore 😄. Bu then the current handling of the Node builtin modules could also be happening done the transformer instead 🤷‍♂️

**Review by devongovett — APPROVED** [review]

**Pull request merged**